packages/squiggle-lang/src/utility/result.ts

```diff
@@ -81,3 +81,13 @@ export function getError<T, E>(r: result<T, E>): E | undefined {
     return undefined;
   }
 }
+
+export function sequence<T, E>(r: result<T, E>[]): result<T[], E> {
```

Comment:
Based on Haskell's `sequence`, also known as `firstErrorOrOpen` in previous code.

Reply:
I started to write a long rant on functions like this last week but ran out of time and lost my draft, and don't really want to spend too much time arguing about functional vs imperative virtues right now.

But, just to state my position for the future when this discussion arises again:
- `firstErrorOrOpen` was a big source of slowdown in our rescript codebase, even though it's not important in this particular case
- dragging around wrapped objects to unpack and repack them later often makes the code more complicated
- in languages without pipes these transformations are more awkward
-on the other hand, in Typescript land we have early returns, which makes it possible to write terse enough code without higher-level functions like this

Overall, I think we should just embrace the imperative style and probably switch to exceptions.